docker/scripts/functions.sh

```diff
@@ -519,6 +519,23 @@ function perc_reset() {
 	fi
 }
 
+function smartarray_reset() {
+	systemmfg=$(dmidecode -s system-manufacturer | head -1)
+	echo "HPE hardware raid device is present on system mfg: $systemmfg"
+
+	slots=$(ssacli ctrl all show detail | awk '/^   Slot: / {print $2}')
+
+	echo "HPE-SSACLI - Deleting all LDs"
+	for slot in $slots; do
+		lds=$(ssacli ctrl slot="$slot" ld all show status | awk '/^   logicaldrive / {print $2}')
+		for ld in $lds; do
+			# ssacli does not have a -yes or -force option that I can find, so yes pipe
+			# slapping `|| true` at the end to sidestep the autofail trap function
+			yes | ssacli ctrl slot="$slot" ld "$ld" delete || true
```

Comment:
I think if we need the `|| true` part, we should have some validation afterwards to make sure all drives were removed after the fact.

Reply:
Ok, looks like we found a better way for deleting the logical drives, so we won't need `|| true`